[subprocess_output] Fix exception raised on empty output + string cmd

olivielpeau opened the pull request:
### What does this PR do?

Without this fix, `_util.get_subprocess_output` wouldn't raise an exception
on empty output, and would instead return `None` (which would then raise
a `TypeError` since we try to use `None` as a tuple).

Fix that.

Also, fix typo for the string cmd logic.

olivielpeau committed:
[subprocess_output] Fix exception raised on empty output + string cmd

Without this fix, `_util.get_subprocess_output` wouldn't raise an exception
on empty output, and would instead return `None` (which would then raise
a `TypeError` since we try to use `None` as a tuple).

Fix that.

Also, fix typo for the string cmd logic.

truthbk commented on pkg/collector/py/datadog_agent.c:
```diff
@@ -86,7 +86,7 @@ static PyObject *get_subprocess_output(PyObject *self, PyObject *args) {
         if (subprocess_arg == NULL) {
             PyErr_SetString(PyExc_Exception, "unable to parse arguments to cgo/go-land");
             free(subprocess_args);
```
we're not releasing the GIL

olivielpeau replied:
fixed 👍

truthbk reviewed (commented):
Thanks for the cleanup. We've got an unreleased GIL, I think.

olivielpeau committed:
Address review comments

We *need* to return NULL to the python interpreter if an exception
has been set. Otherwise the exception is not raised properly in python
land and we're in uncharted undefined behavior land.

olivielpeau committed:
[subprocess_output] Add logging on subprocess

olivielpeau committed:
[subprocess_output] Logging cosmetics

truthbk reviewed (approved):
🚀
